src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -430,52 +430,15 @@ public static SSTableReader openForBatch(Descriptor descriptor, Set<Component> c
             System.exit(1);
         }
 
-        long fileLength = new File(descriptor.filenameFor(Component.DATA)).length();
-        logger.info("Opening {} ({})", descriptor, FBUtilities.prettyPrintMemory(fileLength));
-
-        final SSTableReader sstable;
         try
         {
-            sstable = internalOpen(descriptor,
-                                   components,
-                                   metadata,
-                                   System.currentTimeMillis(),
-                                   statsMetadata,
-                                   OpenReason.NORMAL,
-                                   header.toHeader(metadata.get()));
+            BuilderForBatch builder = new BuilderForBatch(descriptor, metadata);
+            return builder.build(components, statsMetadata, OpenReason.NORMAL, header.toHeader(builder.metadata));
```

Comment:
aside: It's kind of nice now that we don't have to even worry about creating a reader instance until the things that it depends on are actually ready/available.